libfreerdp/codec/h264_ffmpeg.c

```diff
@@ -415,6 +503,25 @@ static BOOL libavcodec_init(H264_CONTEXT* h264)
 			sys->codecDecoderContext->flags |= AV_CODEC_FLAG_TRUNCATED;
 		}
 
+#ifdef WITH_VAAPI
+
+		if (!sys->hwctx)
+		{
+			int ret = av_hwdevice_ctx_create(&sys->hwctx, AV_HWDEVICE_TYPE_VAAPI, VAAPI_DEVICE, NULL, 0);
+
+			if (ret < 0)
+			{
+				WLog_ERR(TAG, "Could not initialize hw decoder: %s", av_err2str(ret));
+				goto EXCEPTION;
+			}
```

Comment:
here I think we should go to fallback without hardware support. Otherwise that means that if the permissions aren't correct for accessing the DRM device, we won't be able to run at all.